src/mpid/ch3/src/mpid_init.c

```diff
@@ -188,6 +192,13 @@ int MPID_Init(int *argc, char ***argv, int requested, int *provided)
     MPIDI_CH3_DBG_REFCOUNT = MPL_dbg_class_alloc("REFCOUNT", "refcount");
 #endif /* MPL_USE_DBG_LOGGING */
 
+    int g_max_node_id;
+    MPIR_Process.node_map = MPL_malloc(pg->size * sizeof(int), MPL_MEM_ADDRESS);
+    MPIR_NODEMAP_build_nodemap(pg->size, pg_rank, MPIR_Process.node_map, &g_max_node_id);
+    MPIR_Process.node_local_map = MPL_malloc(sizeof(int), MPL_MEM_OTHER);
+    MPIR_NODEMAP_get_local_info(pg_rank, pg->size, MPIR_Process.node_map, &MPIR_Process.local_size,
+                                &MPIR_Process.local_rank, MPIR_Process.node_local_map);
+
```

Comment:
Same note as above. Feel free to cherry-pick https://github.com/pmodels/mpich/commit/bae0a67f7282da844e8a05b4a74733239ab1561c

Reply:
I will rebase on your PR directly once it is merged. After that we can also refactor `MPIDU_Init_shm_init` to take no arguments as all the required information is already provided by `MPIR_pmi_init` into `MPIR_Process`.